features/mbedtls/inc/mbedtls/mbedtls_config_reader.h

```diff
@@ -0,0 +1,457 @@
+#endif
+#ifdef MBEDTLS_CONFIG_CCM
+    #define MBEDTLS_CCM_C
+#endif
+
+#ifdef MBEDTLS_CONFIG_PSK
+    #define MBEDTLS_KEY_EXCHANGE_PSK_ENABLED
+#endif
+
+#ifdef MBEDTLS_CONFIG_ECDH
+    #define MBEDTLS_X509_USE_C
+    #define MBEDTLS_PEM_PARSE_C
+    #define MBEDTLS_BASE64_C
+    #define MBEDTLS_ASN1_PARSE_C
+    #define MBEDTLS_PK_PARSE_C
+    #define MBEDTLS_PK_C
+    #define MBEDTLS_ECDH_C
+    #define MBEDTLS_ECP_C
+    #define MBEDTLS_BIGNUM_C
+    #define MBEDTLS_ECP_DP_SECP256R1_ENABLED
+    #define MBEDTLS_ECP_DP_SECP384R1_ENABLED
+    #define MBEDTLS_ECP_DP_CURVE25519_ENABLED
+    #define MBEDTLS_ECP_NIST_OPTIM
+#endif
+
+#ifdef MBEDTLS_CONFIG_ECDH_ECDSA_KEY_EXCHANGE
+    #define MBEDTLS_KEY_EXCHANGE_ECDH_ECDSA_ENABLED
+    #define MBEDTLS_X509_CRT_PARSE_C
+    #define MBEDTLS_SSL_SERVER_NAME_INDICATION
+    #define MBEDTLS_X509_USE_C
+    #define MBEDTLS_PEM_PARSE_C
+    #define MBEDTLS_BASE64_C
+#endif
+
+#ifdef MBEDTLS_CONFIG_RSA
+    #define MBEDTLS_RSA_C
+    #define MBEDTLS_BIGNUM_C
+    #define MBEDTLS_OID_C
+#endif
+
+#ifdef MBEDTLS_CONFIG_ECDHE_RSA_KEY_EXCHANGE
+    #define MBEDTLS_KEY_EXCHANGE_ECDHE_RSA_ENABLED
+    #define MBEDTLS_ECDH_C
+    #define MBEDTLS_RSA_C
+    #define MBEDTLS_PKCS1_V15
+    #define MBEDTLS_PKCS1_V21
+    #define MBEDTLS_X509_CRT_PARSE_C
+    #define MBEDTLS_SSL_SERVER_NAME_INDICATION
+    #define MBEDTLS_X509_USE_C
+    #define MBEDTLS_PEM_PARSE_C
+    #define MBEDTLS_BASE64_C
+#endif
+
+#ifdef MBEDTLS_CONFIG_RSA_KEY_EXCHANGE
+    #define MBEDTLS_KEY_EXCHANGE_RSA_ENABLED
+    #define MBEDTLS_RSA_C
+    
+    #define MBEDTLS_PKCS1_V15
+    #define MBEDTLS_X509_CRT_PARSE_C
+    #define MBEDTLS_SSL_SERVER_NAME_INDICATION
+    #define MBEDTLS_X509_USE_C
+    #define MBEDTLS_PEM_PARSE_C
+    #define MBEDTLS_BASE64_C
+    #define MBEDTLS_BIGNUM_C
+    #define MBEDTLS_OID_C
+    #define MBEDTLS_ASN1_PARSE_C
+    #define MBEDTLS_PK_PARSE_C
+    #define MBEDTLS_PK_C
+#endif
+
+#ifdef MBEDTLS_CONFIG_ECDH_RSA_KEY_EXCHANGE
+    #define MBEDTLS_KEY_EXCHANGE_ECDH_RSA_ENABLED
+    #define MBEDTLS_PKCS1_V15
+    #define MBEDTLS_ECDH_C
+    #define MBEDTLS_X509_CRT_PARSE_C
+    #define MBEDTLS_SSL_SERVER_NAME_INDICATION
+#endif
+
+#define MBED_TLS_HAVE_ASM
+#define MBED_TLS_HAVE_TIME
+#define MBEDTLS_ENTROPY_C
+#define MBEDTLS_SHA256_C
+#define MBEDTLS_HAVE_TIME
+#define MBEDTLS_REMOVE_ARC4_CIPHERSUITES
+#define MBEDTLS_SSL_TLS_C
+#define MBEDTLS_CIPHER_C
+#define MBEDTLS_MD_C
+#define MBEDTLS_ERROR_STRERROR_DUMMY
+#define MBEDTLS_NO_PLATFORM_ENTROPY
+#define MBEDTLS_SSL_EXTENDED_MASTER_SECRET
+#define MBEDTLS_SSL_PROTO_TLS1_2
+#define MBEDTLS_X509_CHECK_KEY_USAGE
+#define MBEDTLS_X509_CHECK_EXTENDED_KEY_USAGE
+#define MBEDTLS_VERSION_C
+#define MBEDTLS_SELF_TEST
+#define MBEDTLS_PK_RSA_ALT_SUPPORT
+#define MBEDTLS_SSL_ALL_ALERT_MESSAGES
+#define MBEDTLS_SSL_ENCRYPT_THEN_MAC
+#define MBEDTLS_SSL_RENEGOTIATION
+#define MBEDTLS_SSL_MAX_FRAGMENT_LENGTH
+#define MBEDTLS_SSL_PROTO_DTLS
```

Comment:
Do we want to support DTLS by defualt?

Reply:
That's what's defined in the `config.h` file in `mbed-os/features/mbedtls/inc/mbedtls/`.
This configuration is designed to default as close as possible to the existing MbedTLS configuration in Mbed OS.

Reply:
OK